node-graph/gcore/src/vector/vector_nodes.rs

```diff
@@ -1135,65 +1135,243 @@ fn bevel_algorithm(mut vector_data: VectorData, distance: f64) -> VectorData {
+
+					if i > 0 {
+						let prev_sample = max_split * ((i - 1) as f64 / INITIAL_SAMPLES as f64);
+
+						const REFINE_STEPS: usize = 10;
+						for j in 1..=REFINE_STEPS {
+							let refined_sample = prev_sample + (distance_sample - prev_sample) * (j as f64 / REFINE_STEPS as f64);
+
+							let x_point = bezier1.evaluate(TValue::Euclidean(1.0 - (refined_sample / length1).clamp(0.0, 1.0)));
+							let y_point = bezier2.evaluate(TValue::Euclidean((refined_sample / length2).clamp(0.0, 1.0)));
+
+							let distance = x_point.distance(y_point);
+
+							if bevel_length - distance < 0.0 {
+								split_distance = refined_sample;
+								break;
+							}
+						}
+					}
+					break;
+				}
+			}
+
+			if split_distance == 0.0 && current_best_distance > 0.0 {
+				split_distance = current_best_distance;
+			}
+
+			split_distance
+		}
+	}
+
 	fn update_existing_segments(vector_data: &mut VectorData, distance: f64, segments_connected: &mut [u8]) -> Vec<[usize; 2]> {
 		let mut next_id = vector_data.point_domain.next_id();
 		let mut new_segments = Vec::new();
 
-		for (handles, start_point_index, end_point_index) in vector_data.segment_domain.handles_and_points_mut() {
-			// Convert the original segment to a bezier
+		let mut first_bezier_bool = true;
+
+		let mut iter = vector_data.segment_domain.handles_and_points_mut();
+
+		let mut first_bezier = bezier_rs::Bezier {
+			end: DVec2 { x: 0., y: 0. },
+			start: DVec2 { x: 0., y: 0. },
+			handles: bezier_rs::BezierHandles::Linear,
+		};
+
+		let mut first_start_point_index = &mut 0xffffffff;
+		let mut first_handles = &mut bezier_rs::BezierHandles::Linear;
+
+		let Some((handles, start_point_index, end_point_index)) = iter.next() else { unreachable!() };
+
+		let mut prev_bezier = bezier_rs::Bezier {
+			start: vector_data.point_domain.positions()[*start_point_index],
+			end: vector_data.point_domain.positions()[*end_point_index],
+			handles: *handles,
+		};
+
+		let mut prev_handles = handles;
+		let mut prev_start_point_index = start_point_index;
+		let mut prev_end_point_index = end_point_index;
+		let mut prev_original_length = prev_bezier.length(None);
+		let mut prev_length = prev_original_length;
+		let mut first_original_length = prev_original_length;
+		let mut first_length = prev_original_length;
+
+		while let Some((handles, start_point_index, end_point_index)) = iter.next() {
 			let mut bezier = bezier_rs::Bezier {
 				start: vector_data.point_domain.positions()[*start_point_index],
 				end: vector_data.point_domain.positions()[*end_point_index],
 				handles: *handles,
 			};
 
+			let spilt_distance = calculate_distance_to_spilt(prev_bezier, bezier, distance);
+
+			if prev_bezier.is_linear() {
+				prev_bezier.handles = bezier_rs::BezierHandles::Linear;
+			}
+
 			if bezier.is_linear() {
 				bezier.handles = bezier_rs::BezierHandles::Linear;
 			}
+
+			prev_bezier = prev_bezier.apply_transformation(|p| vector_data.transform.transform_point2(p));
```

Comment:
This doesn't work when the layer is transformed:
https://github.com/user-attachments/assets/7a8f031f-9b6a-475d-a890-7999c4161b92